`./scripts/run_checks.sh` fails locally

Running autoformatting.
Skipped 2 files
Autoformatting complete.
Running type checking.
src/envs/doors.py:430: error: Call to untyped function "norm" in typed context
src/envs/doors.py:537: error: Call to untyped function "norm" in typed context
Found 2 errors in 1 file (checked 131 source files)
Type checking failed! Terminating check script early.

--- a/requirements.txt
+++ b/requirements.txt
@@ -1,4 +1,4 @@
-numpy>=1.21.0
+numpy>=1.22.3
 pytest
 gym==0.21.0
 matplotlib

--- a/src/approaches/gnn_metacontroller_approach.py
+++ b/src/approaches/gnn_metacontroller_approach.py
@@ -167,7 +167,7 @@ class GNNMetacontrollerApproach(NSRTLearningApproach, GNNApproach):
             for j in range(len(scores)):
                 if j not in allowed_idxs:
                     scores[j] = float("-inf")  # set its score to be really bad
-            if np.max(scores) == float("-inf"):  # type: ignore
+            if np.max(scores) == float("-inf"):
                 # If all scores are -inf, we failed to select an object.
                 raise ApproachFailure(
                     "GNN metacontroller could not select an object")

--- a/src/approaches/gnn_option_policy_approach.py
+++ b/src/approaches/gnn_option_policy_approach.py
@@ -143,8 +143,8 @@ class GNNOptionPolicyApproach(GNNApproach):
         for obj, node in object_to_node.items():
             type_to_node[obj.type.name].add(node)
         # Extract parameterized option and continuous parameters.
-        onehot_output, params = np.split(  # type: ignore
-            graph_output["globals"], [len(self._sorted_options)])
+        onehot_output, params = np.split(graph_output["globals"],
+                                         [len(self._sorted_options)])
         param_opt = self._sorted_options[np.argmax(onehot_output)]
         # Pad and clip parameters.
         params = params[:param_opt.params_space.shape[0]]
@@ -158,7 +158,7 @@ class GNNOptionPolicyApproach(GNNApproach):
             for j in range(len(scores)):
                 if j not in allowed_idxs:
                     scores[j] = float("-inf")  # set its score to be really bad
-            if np.max(scores) == float("-inf"):  # type: ignore
+            if np.max(scores) == float("-inf"):
                 # If all scores are -inf, we failed to select an object.
                 raise ApproachFailure(
                     "GNN option policy could not select an object")

--- a/src/envs/behavior_options.py
+++ b/src/envs/behavior_options.py
@@ -439,7 +439,7 @@ def get_delta_low_level_hand_action(
     )
 
     delta_trig_frac = 0
-    action = np.concatenate(  # type: ignore
+    action = np.concatenate(
         [
             np.zeros((10), dtype=np.float32),
             np.array(delta_pos, dtype=np.float32),
@@ -775,7 +775,7 @@ def grasp_obj_at_pos(
         return None
 
     # If the object is too far away, fail and return None
-    if (np.linalg.norm(  # type: ignore
+    if (np.linalg.norm(
             np.array(obj.get_position()) -
             np.array(env.robots[0].get_position())) > 2):
         logging.info(f"PRIMITIVE: grasp {obj.name} fail, too far")
@@ -809,24 +809,21 @@ def grasp_obj_at_pos(
     hand_to_obj_unit_vector = hand_to_obj_vector / \
         np.linalg.norm(
         hand_to_obj_vector
-    )  # type: ignore
+    )
     unit_z_vector = np.array([0.0, 0.0, -1.0])
     # This is because we assume the hand is originally oriented
     # so -z is coming out of the palm
-    c_var = np.dot(
-        unit_z_vector,  # type: ignore
-        hand_to_obj_unit_vector)
+    c_var = np.dot(unit_z_vector, hand_to_obj_unit_vector)
     if c_var not in [-1.0, 1.0]:
         v_var = np.cross(unit_z_vector, hand_to_obj_unit_vector)
-        s_var = np.linalg.norm(v_var)  # type: ignore
+        s_var = np.linalg.norm(v_var)
         v_x = np.array([
             [0, -v_var[2], v_var[1]],
             [v_var[2], 0, -v_var[0]],
             [-v_var[1], v_var[0], 0],
         ])
-        R = (
-            np.eye(3) + v_x + np.linalg.matrix_power(v_x, 2)  # type: ignore
-            * ((1 - c_var) / (s_var**2)))
+        R = (np.eye(3) + v_x + np.linalg.matrix_power(v_x, 2) * ((1 - c_var) /
+                                                                 (s_var**2)))
         r = scipy.spatial.transform.Rotation.from_matrix(R)
         euler_angles = r.as_euler("xyz")
     else:

--- a/src/envs/blocks.py
+++ b/src/envs/blocks.py
@@ -486,9 +486,7 @@ class BlocksEnv(BaseEnv):
                 state.get(block, "pose_z")
             ])
             if np.allclose([x, y, z], block_pose, atol=self.pick_tol):
-                dist = np.linalg.norm(
-                    np.array([x, y, z]) -  # type: ignore
-                    block_pose)
+                dist = np.linalg.norm(np.array([x, y, z]) - block_pose)
                 close_blocks.append((block, float(dist)))
         if not close_blocks:
             return None

--- a/src/envs/cluttered_table.py
+++ b/src/envs/cluttered_table.py
@@ -72,8 +72,7 @@ class ClutteredTableEnv(BaseEnv):
             this_x = state.get(can, "pose_x")
             this_y = state.get(can, "pose_y")
             this_radius = state.get(can, "radius")
-            if np.linalg.norm([end_x - this_x, end_y - this_y
-                               ]) < this_radius:  # type: ignore
+            if np.linalg.norm([end_x - this_x, end_y - this_y]) < this_radius:
                 assert desired_can is None
                 desired_can = can
         if desired_can is None:
@@ -242,8 +241,7 @@ class ClutteredTableEnv(BaseEnv):
             other_x = other_feats[0]
             other_y = other_feats[1]
             other_radius = other_feats[2]
-            distance = np.linalg.norm([other_x - pose[0],
-                                       other_y - pose[1]])  # type: ignore
+            distance = np.linalg.norm([other_x - pose[0], other_y - pose[1]])
             if distance <= (radius + other_radius):
                 return True
         return False
@@ -276,13 +274,10 @@ class ClutteredTableEnv(BaseEnv):
             vec2 = np.array([end_x - this_x, end_y - this_y])
             angle = np.arccos(
                 np.clip(
-                    vec1.dot(vec2) / (
-                        np.linalg.norm(vec1) *  # type: ignore
-                        np.linalg.norm(vec2)),  # type: ignore
-                    -1.0,
-                    1.0))
+                    vec1.dot(vec2) /
+                    (np.linalg.norm(vec1) * np.linalg.norm(vec2)), -1.0, 1.0))
             if abs(angle) < CFG.cluttered_table_collision_angle_thresh:
-                dist = np.linalg.norm(vec2)  # type: ignore
+                dist = np.linalg.norm(vec2)
                 if dist > colliding_can_max_dist:
                     colliding_can_max_dist = float(dist)
                     colliding_can = can
@@ -363,8 +358,7 @@ class ClutteredTablePlaceEnv(ClutteredTableEnv):
             this_x = state.get(can, "pose_x")
             this_y = state.get(can, "pose_y")
             this_radius = state.get(can, "radius")
-            if np.linalg.norm([end_x - this_x, end_y - this_y
-                               ]) < this_radius:  # type: ignore
+            if np.linalg.norm([end_x - this_x, end_y - this_y]) < this_radius:
                 assert desired_can is None
                 desired_can = can
         if desired_can is None:

--- a/src/envs/cover.py
+++ b/src/envs/cover.py
@@ -1002,12 +1002,8 @@ class CoverMultistepOptions(CoverEnvTypedOptions):
         m["absolute_params"] = vec + p
         return self._HandEmpty_holds(s, [])
 
-    def _Pick_policy(
-            self,
-            s: State,
-            m: Dict,  # type: ignore
-            o: Sequence[Object],
-            p: Array) -> Action:
+    def _Pick_policy(self, s: State, m: Dict, o: Sequence[Object],
+                     p: Array) -> Action:
         assert np.allclose(p, m["params"])
         del p
         absolute_params = m["absolute_params"]

--- a/src/envs/pybullet_env.py
+++ b/src/envs/pybullet_env.py
@@ -296,9 +296,7 @@ class PyBulletEnv(BaseEnv):
         closest_held_obj_dist = float("inf")
         for obj_id in self._get_object_ids_for_held_check():
             for finger_id, expected_normal in expected_finger_normals.items():
-                assert abs(
-                    np.linalg.norm(expected_normal) -  # type: ignore
-                    1.0) < 1e-5
+                assert abs(np.linalg.norm(expected_normal) - 1.0) < 1e-5
                 # Find points on the object that are within grasp_tol distance
                 # of the finger. Note that we use getClosestPoints instead of
                 # getContactPoints because we still want to consider the object

--- a/src/envs/pybullet_robots.py
+++ b/src/envs/pybullet_robots.py
@@ -414,7 +414,7 @@ def create_move_end_effector_to_pose_option(
         # Run IK to determine the target joint positions.
         ee_delta = np.subtract(target, current)
         # Reduce the target to conform to the max velocity constraint.
-        ee_norm = np.linalg.norm(ee_delta)  # type: ignore
+        ee_norm = np.linalg.norm(ee_delta)
         if ee_norm > max_vel_norm:
             ee_delta = ee_delta * max_vel_norm / ee_norm
         ee_action = np.add(current, ee_delta)

--- a/src/gnn/gnn_utils.py
+++ b/src/gnn/gnn_utils.py
@@ -135,8 +135,8 @@ def compute_normalizers(data: List[Dict]) -> Dict[str, Tuple[Array, Array]]:
 
 def _compute_normalizer_array(array_data: Array) -> Tuple[Array, Array]:
     """Helper function for compute_normalizers()."""
-    shift = np.min(array_data, axis=0)  # type: ignore
-    scale = np.max(array_data - shift, axis=0)  # type: ignore
+    shift = np.min(array_data, axis=0)
+    scale = np.max(array_data - shift, axis=0)
     shift[np.where(scale == 0)] = 0
     scale[np.where(scale == 0)] = 1
     return shift, scale
@@ -232,7 +232,7 @@ def _unstack(array: Array) -> List[torch.Tensor]:
     num_splits = int(array.shape[0])
     return [
         torch.squeeze(x, dim=0)  # type: ignore
-        for x in np.split(array, num_splits, axis=0)  # type: ignore
+        for x in np.split(array, num_splits, axis=0)
     ]
 
 
@@ -254,16 +254,14 @@ def split_graphs(graph: Dict) -> List[Dict]:
     edges_splits = np.cumsum(graph['n_edge'][:-1])
     graph_of_lists: Dict[str, Any] = collections.defaultdict(lambda: [])
     if graph['nodes'] is not None:
-        graph_of_lists['nodes'] = np.split(  # type: ignore
-            graph['nodes'], nodes_splits)
+        graph_of_lists['nodes'] = np.split(graph['nodes'], nodes_splits)
     if graph['edges'] is not None:
-        graph_of_lists['edges'] = np.split(  # type: ignore
-            graph['edges'], edges_splits)
+        graph_of_lists['edges'] = np.split(graph['edges'], edges_splits)
     if graph['receivers'] is not None:
-        graph_of_lists['receivers'] = np.split(  # type: ignore
-            graph['receivers'] - offset, edges_splits)
-        graph_of_lists['senders'] = np.split(  # type: ignore
-            graph['senders'] - offset, edges_splits)
+        graph_of_lists['receivers'] = np.split(graph['receivers'] - offset,
+                                               edges_splits)
+        graph_of_lists['senders'] = np.split(graph['senders'] - offset,
+                                             edges_splits)
     if graph['globals'] is not None:
         graph_of_lists['globals'] = _unstack(graph['globals'])
 

--- a/src/ml_models.py
+++ b/src/ml_models.py
@@ -176,7 +176,7 @@ class PyTorchRegressor(_NormalizingRegressor, nn.Module):
         tensor_X = tensor_x.unsqueeze(dim=0)
         tensor_Y = self(tensor_X)
         tensor_y = tensor_Y.squeeze(dim=0)
-        y = tensor_y.detach().numpy()  # type: ignore
+        y = tensor_y.detach().numpy()
         return y
 
 
@@ -380,7 +380,7 @@ class PyTorchBinaryClassifier(_NormalizingBinaryClassifier, nn.Module):
         tensor_X = tensor_x.unsqueeze(dim=0)
         tensor_Y = self(tensor_X)
         tensor_y = tensor_Y.squeeze(dim=0)
-        y = tensor_y.detach().numpy()  # type: ignore
+        y = tensor_y.detach().numpy()
         proba = y.item()
         assert 0 <= proba <= 1
         return proba
@@ -668,7 +668,7 @@ class ImplicitMLPRegressor(PyTorchRegressor):
         assert self._grid_num_ticks_per_dim > 0
         dy = 1.0 / self._grid_num_ticks_per_dim
         ticks = [np.arange(0.0, 1.0, dy)] * self._y_dim
-        grid = np.meshgrid(*ticks)  # type: ignore
+        grid = np.meshgrid(*ticks)
         candidate_ys = np.transpose(grid).reshape((-1, self._y_dim))
         num_samples = candidate_ys.shape[0]
         assert num_samples == self._grid_num_ticks_per_dim**self._y_dim
@@ -880,8 +880,8 @@ class LearnedPredicateClassifier:
 
 def _normalize_data(data: Array,
                     scale_clip: float = 1) -> Tuple[Array, Array, Array]:
-    shift = np.min(data, axis=0)  # type: ignore
-    scale = np.max(data - shift, axis=0)  # type: ignore
+    shift = np.min(data, axis=0)
+    scale = np.max(data - shift, axis=0)
     scale = np.clip(scale, scale_clip, None)
     return (data - shift) / scale, shift, scale
 

--- a/src/utils.py
+++ b/src/utils.py
@@ -470,7 +470,7 @@ def line_segment_intersects_circle(seg: LineSegment,
     b = (seg.x2, seg.y2)
     ba = np.subtract(b, a)
     ca = np.subtract(c, a)
-    da = ba * np.dot(ca, ba) / np.dot(ba, ba)  # type: ignore
+    da = ba * np.dot(ca, ba) / np.dot(ba, ba)
     # The point on the extended line that is the closest to the center.
     d = dx, dy = (a[0] + da[0], a[1] + da[1])
     # Optionally plot the important points.
@@ -2098,9 +2098,7 @@ def fig2data(fig: matplotlib.figure.Figure, dpi: int = 150) -> Image:
     """Convert matplotlib figure into Image."""
     fig.set_dpi(dpi)
     fig.canvas.draw()
-    data = np.frombuffer(
-        fig.canvas.tostring_argb(),  # type: ignore
-        dtype=np.uint8).copy()
+    data = np.frombuffer(fig.canvas.tostring_argb(), dtype=np.uint8).copy()
     data = data.reshape(fig.canvas.get_width_height()[::-1] + (4, ))
     data[..., [0, 1, 2, 3]] = data[..., [1, 2, 3, 0]]
     return data

